jgrapht-core/src/main/java/org/jgrapht/alg/spanning/PrimMinimumSpanningTree.java

```diff
@@ -103,6 +103,91 @@ public PrimMinimumSpanningTree(Graph<V, E> graph)
 
         return new SpanningTreeImpl<>(minimumSpanningTreeEdgeSet, spanningTreeWeight);
     }
+
+
+    /**
+     * Computes a spanning tree in $O(|V|^2)$ time.
```

Comment:
Can you add a reference to the source you used for implementation? How does it differ from the other implementation of Prim?

Reply:
I don't think there exists a reference I can add. The algorithm is the first one described under Time complexity on https://en.wikipedia.org/wiki/Prim%27s_algorithm.

The other implementation uses a heap and works in O(|E| log |V|) while this ones only uses a couple of arrays and runs in O(|V|^2).